### PR TITLE
*: snapshot doesn't contain logs.

### PR DESCRIPTION
**Pull request by siddontang**
@ngaut @BusyJay

### CODE REVIEW

**Comment by ngaut** [comment]
LGTM

File: src/raftstore/store/peer_storage.rs [hunk 1]
```diff
@@ -353,12 +363,20 @@ impl PeerStorage {
         let last_index = snap.get_metadata().get_index();
         try!(save_last_index(w, region_id, last_index));
 
+        // The snapshot only contains log which index > applied index, so
+        // here the truncate state's (index, term) is in snapshot metadata.
+        let mut truncated_state = RaftTruncatedState::new();
+        truncated_state.set_index(last_index);
```
Comment (BusyJay) [inline src/raftstore/store/peer_storage.rs]: Maybe we should alter gc check in store.

**Comment by BusyJay** [comment]
rest LGTM

**Comment by BusyJay** [comment]
LGTM

**Pull request merged**